src/cpp/session/modules/SessionPackages.cpp

```diff
@@ -537,44 +546,22 @@ Error initialize()
 {
    // register deferred init
    module_context::events().onDeferredInit.connect(onDeferredInit);
-
-   // register routines
-   R_CallMethodDef methodDef ;
-   methodDef.name = "rs_enqueLoadedPackageUpdates" ;
-   methodDef.fun = (DL_FUNC) rs_enqueLoadedPackageUpdates ;
-   methodDef.numArgs = 1;
-   r::routines::addCallMethod(methodDef);
-
-   R_CallMethodDef methodDef2 ;
-   methodDef2.name = "rs_canInstallPackages" ;
-   methodDef2.fun = (DL_FUNC) rs_canInstallPackages ;
-   methodDef2.numArgs = 0;
-   r::routines::addCallMethod(methodDef2);
-
-   R_CallMethodDef methodDef3 ;
-   methodDef3.name = "rs_packageLibraryMutated" ;
-   methodDef3.fun = (DL_FUNC) rs_packageLibraryMutated ;
-   methodDef3.numArgs = 0;
-   r::routines::addCallMethod(methodDef3);
-   
-   r::routines::registerCallMethod(
-            "rs_getCachedAvailablePackages",
-            (DL_FUNC) rs_getCachedAvailablePackages,
-            1);
-   
-   r::routines::registerCallMethod(
-            "rs_downloadAvailablePackages",
-            (DL_FUNC) rs_downloadAvailablePackages,
-            1);
    
+   RS_REGISTER_CALL_METHOD(rs_enqueLoadedPackageUpdates, 1);
+   RS_REGISTER_CALL_METHOD(rs_canInstallPackages, 0);
+   RS_REGISTER_CALL_METHOD(rs_packageLibraryMutated, 0);
+   RS_REGISTER_CALL_METHOD(rs_getCachedAvailablePackages, 1);
+   RS_REGISTER_CALL_METHOD(rs_downloadAvailablePackages, 1);
+   RS_REGISTER_CALL_METHOD(rs_packagesPaneEnabled, 0);
+
    using boost::bind;
    using namespace module_context;
    ExecBlock initBlock ;
    initBlock.addFunctions()
       (bind(sourceModuleRFile, "SessionPackages.R"))
       (bind(registerRpcMethod, "available_packages", availablePackages))
-      (bind(registerRpcMethod, "get_package_state", getPackageState))
-      (bind(r::exec::executeString, ".rs.packages.initialize()"));
```

Comment:
@jjallaire, I moved this to deferred init because otherwise this code tries to run before the native routines have actually been registered (and so we can't successfully check `.rs.routines$rs_packagesPaneEnabled`).

Do you know if there's any code that depends on `.rs.packages.initialize()` running eagerly or do you think a deferred init should be okay? (Since it just adds a bunch of hooks / handlers it seems like a deferred init should be okay)